synapse/handlers/receipts.py

```diff
@@ -188,15 +188,22 @@ def filter_out_hidden(events: List[JsonDict], user_id: str) -> List[JsonDict]:
 
                 new_users = {}
                 for rr_user_id, user_rr in m_read.items():
-                    hidden = user_rr.get("hidden", None)
-                    if hidden is not True or rr_user_id == user_id:
-                        new_users[rr_user_id] = user_rr.copy()
-                        # If hidden has a value replace hidden with the correct prefixed key
-                        if hidden is not None:
-                            new_users[rr_user_id].pop("hidden")
-                            new_users[rr_user_id][
-                                ReadReceiptEventFields.MSC2285_HIDDEN
-                            ] = hidden
+                    try:
+                        hidden = user_rr.get("hidden", None)
```

Comment:
```suggestion
                        hidden = user_rr.get("hidden")
```

`None` is the default value for the second argument of `get` (might as well fix while we're here).